main.tex

```diff
@@ -578,105 +732,156 @@ \subsubsection{Malicious Coordinator}
+
+\section{Performance}\label{sec:performanceanalysis}
+
+In this section we discuss the efficiency of our protocol. Because the protocol can only complete successfully when all participants are both honest and available, and due to the underlying anonymous overlay network it is desirable to operate with as little communication complexity as possible. Reducing communication costs can also make the protocol more robust against traffic analysis by a global passive adversary or targeted censorship by an active one. Computational complexity is mainly a consideration for the coordinator it must process the requests of all users, which follow a bursty pattern.
+
+\subsection{Theoretical performance}
+
+The following table estimates communication overheads for the protocol assuming simple $\Sigma$-protocol for linear relations~\cite[19.5.3 pp. 747-8, 20.4.1 pp. 792]{boneh2020graduate}. These proofs are amenable to compression~\cite{bunz2018bulletproofs, attema2020compressed} and could be reduced to a size that is logarithmic in the bit width and the number of credentials per request.
+
+\begin{table*}[h!]
+\centering
+\begin{tabular}{l|cccccc|cc}
+                 & \multicolumn{6}{c|}{Request}                                                                                            & \multicolumn{2}{c}{Response}   \\
+                 & $M_a$ & $\pi^{\textit{range}}$ & $C_a$ & $S$ & $(\pi^{\textit{serial}} \land \pi^{\textit{MAC}})
+                 $ & $\pi^{\textit{sum}}$ & MAC & $\pi^{\textit{iparams}}$ \\
+$\#\mathbb{G}$   & $k$   & $k(2n + 1)$            & $k$   & $k$ & $4k$                                             & $1$                  & $k$ & $3k$                     \\
+$\#\mathbb{Z}_q$ & $-$   & $k(3n + 1)$            & $-$   & $-$ & $5k$                                             & $2$                  & $k$ & $5k$
+\end{tabular}
+
+  \caption{Communication overhead of the WabiSabi protocol. Each round trip corresponds to a single registered input and $k$ denotes the number of requested credentials. The bit-width of the range proofs is denoted as $n$, and for bootstrap requests its value is zero.}\label{tab:protocolefficiency}
+\end{table*}
+
+\subsection{Concrete performance}
+
+A proof of concept implementation of the credential system has been added to the open source Wasabi wallet\footnote{\url{https://github.com/zkSNACKs/WalletWasabi}}. This implementation relied on \texttt{NBitcoin.Secp256k1}\footnote{\url{https://github.com/MetacoSA/NBitcoin/tree/master/NBitcoin.Secp256k1}} a port of \texttt{libsecp256k1}\footnote{\url{https://github.com/bitcoin-core/secp256k1/pull/486}}. Due to limitations of the .NET runtime only 32-bit limb field elements are supported. All proofs were made non-interactive using the strong variant of the Fiat-Shamir transform~\cite{fiat1987prove, bernhard2012not, hamburg2017strobe}.
```

Comment:
> Due to limitations of the .NET runtime only 32-bit limb field elements are supported

What are these limitations? Without a link this sounds like "Due my dog has eaten my homework".

Reply:
64 bit limbs require a 128 bit integer type which is not available in CLR, so even though the hardware is capable of using 64 bit limbs and this is the default for 64 bit builds of libsecp256k1, this is not possible in NBitcoin